packages/dataviews/src/types.ts

```diff
@@ -292,24 +292,36 @@ interface ViewBase {
 	 * The fields to render
 	 */
 	fields?: string[];
-}
 
-export interface CombinedField {
-	id: string;
+	/**
+	 * Title field
+	 */
+	titleField?: string;
 
-	label: string;
+	/**
+	 * Media field
+	 */
+	mediaField?: string;
 
-	header?: string | ReactElement;
+	/**
+	 * Description field
+	 */
+	descriptionField?: string;
 
 	/**
-	 * The fields to use as columns.
+	 * Whether to show the title
 	 */
-	children: string[];
+	showTitle?: boolean;
 
 	/**
-	 * The direction of the stack.
+	 * Whether to show the media
 	 */
-	direction: 'horizontal' | 'vertical';
+	showMedia?: boolean;
```

Comment:
Could we remove the showMedia Property and assume that if mediaField is not on the list of visible fields it is hidden?
Related soon we would have the ability to offer multiple media fields and switch between them https://github.com/WordPress/gutenberg/pull/67278.

Reply:
> Could we remove the showMedia Property and assume that if mediaField is not on the list of visible fields it is hidden?

If we do this, the "reordering" of the fields (other than media, title, description) becomes more complex, we'd need to "skip" them kind of regardless of where they are in the array.

Reply:
I guess it is ok to keep showMedia for now but on https://github.com/WordPress/gutenberg/pull/67278 we may need to do changes.

Reply:
Why would we change it?

Reply:
I think these two things might be separate things. "picking a media field" and "toggling its visibility". We have `mediaField` property in the `view` for picking the mediaField and `showMedia` is for whether to show it or not.

Reply:
The plan was for mediaField to become mediaFields to indicate the fields that could be media. We could say the first array item is the active one.

One thing we are not able to express would be to allow multiple media fields to be shown on the table layout, but I guess that's okay; otherwise, we would have different behaviors between the layouts.

Reply:
I think the active media field and the available media fields are two different things. For the latter I'm not sure it's something for the "view" object and more a config of the fields themselves.

Reply:
> For the latter I'm not sure it's something for the "view" object and more a config of the fields themselves.

We were discussing that at https://github.com/WordPress/gutenberg/pull/67278#discussion_r1865668433.
We can continue the discussion there both options are a possibility either the field it self says it is media, or the view says which fields are media. I guess we need to decide.